src/main/flight/gps_rescue.c

```diff
@@ -529,12 +529,10 @@ void updateGPSRescueState(void)
             rescueState.failure = RESCUE_TOO_CLOSE;
             
             // Never allow rescue mode to engage as a failsafe when too close.
-            if (rescueState.isFailsafe) {
+            if (rescueState.isFailsafe || isModeActivationConditionPresent(BOXGPSRESCUE)) {
```

Comment:
Not quite sure if this is really needed - one could argue that when GPS rescue is initialised by switch then it can be aborted at any stage, so ignoring the minimum distance here should be ok.

Reply:
Without this, user with `RESCUE_SANITY_OFF` can enable gps rescue at any distance and potentially hurt himself

Reply:
Agreed. But if they enable GPS rescue they can just disable it again before this happens, so this is no different from a user willingly flying into themselves while manually controlling the craft.
I am not saying that this is absolutely wrong, I am just trying to weigh up the options between safety (that is not such a big concern if the user is still in control of the craft) and the ability to test / tune GPS rescue without minimum distance requirements.

Reply:
The only difference is that the user can accidenttally hit the gps rescue switch and the quad will accelerate at throttleMax setting(1600 by default), I don't think that testing gps rescue from home point is very useful. For me the current behavior with `RESCUE_SANITY_OFF` is a bit dangerous.

Reply:
Another thing is the setting consistency, for me `minRescueDth` needs to works in the same way with all the sanity check types, and not to works yes or no (without any wiki that explains this behavior). IMHO

Reply:
> The only difference is that the user can accidenttally hit the gps rescue switch

This is no different from the pilot accidentally hitting the throttle stick - both may result in the craft accelerating fast towards them.

> Another thing is the setting consistency, for me `minRescueDth` needs to work in the same way with all the sanity check types,

`minRescueDth` already works different to every other santity check, as in it is observed in the case of a failsafe even if sanity checking is off. Now you are proposing to make it even more different, by observing it with sanity checks off even if it is the pilot initiating the GPS rescue.

Reply:
I know `minRescueDth` already works different to every other sanity checks and I'm proposing to make it the same over the different sanity checks

Reply:
In a failsafe, if less than  `minRescueDth` if I'm not wrong the craft is disarmed. To make it work the same, are you going to disarm the craft if the user calls the GPS Rescue with a switch?

Reply:
With current default settings (`RESCUE_SANITY_ON`)  if the distance to home is less than minRescueDth the craft disarms both in a failsafe and with the switch.

With `RESCUE_SANITY_OFF` and `RESCUE_SANITY_FS_ONLY`  if the distance to home is less than minRescueDth the craft disarms only in a failsafe and will ignore `minRescueDth` when gps rescue is engaged by switch.

Reply:
> I know minRescueDth already works different to every other sanity checks and I'm proposing to make it the same over the different sanity checks

But why?

I am not sure that 'for consistency' is a good argument if it takes away user options without fixing an actual problem.

Reply:
@IllusionFpv With respect. I'll give a test example of when sanity check could be problem.
I was trying to balance `min_throttle` and `hover_throttle `between a 3S and 4S setup, so that GPS rescue would come in slower when approaching the home location. (shouldn't be a problem anymore with your vertical velocity PID PR). :+1: 
What happened. Was I had the `Min_throttle/Hover_throttle`set a little to high for 4S. This caused the craft to continue to climb on the way home. Which in turn increased the distance from home, via altitude, although the craft was almost over the home location.
I was lucky to pick up on this quickly, or the craft could have disarmed and fell from a great height... on me !